src/clients/MultiCallBundler.ts

```diff
@@ -17,18 +25,97 @@ export class MultiCallBundler {
     this.transactions = [];
   }
 
+  getTarget(index: number) {
+    return { target: this.transactions[index].contract.address };
+  }
+
   async executeTransactionQueue() {
-    // TODO: this should include grouping logic for multicall
-    this.logger.debug({ at: "MultiCallBundler", message: "Executing tx bundle", number: this.transactions.length });
-    const transactionResults = await Promise.all(this.transactions);
-    const transactionReceipts = await Promise.all(transactionResults.map((transaction: any) => transaction.wait()));
-
-    // TODO: add additional logging on error processing.
-    this.logger.debug({
-      at: "MultiCallBundler",
-      message: "All transactions executed",
-      hashes: transactionReceipts.map((receipt) => receipt.transactionHash),
-    });
-    return transactionReceipts;
+    try {
+      this.logger.debug({ at: "MultiCallBundler", message: "Executing tx bundle", number: this.transactions.length });
+
+      // Simulate the transaction execution for the whole queue.
+      const transactionsSucceed = await Promise.all(
+        this.transactions.map(async (tx: preSignedTransaction) => willSucceed(tx.contract, tx.method, tx.args))
```

Comment:
This should be `await willSucceed` right?
```suggestion
        this.transactions.map(async (tx: preSignedTransaction) => await willSucceed(tx.contract, tx.method, tx.args))
```

Reply:
aha no. the map should not be async seeing this is within a `Promise.all` it must return a promise and not await it.